### PR TITLE
Helpful Script to Advance Time in the EMP 

### PR DESCRIPTION
**Pull request by nicholaspai**
I found that when testing the `request-withdrawal` functionality in the Sponsor CLI I needed to advance time in the EMP. We currently have a script in `/scripts/local/AdvanceToNextVotingPhase.js` which advances time in the DVM, but do not have one for the EMP.

This script allows you to pass in some amount of seconds to advance time by, otherwise it defaults to 3601: 1 hour and 1 second. This amount of time added to a withdrawal request & liquidation request will "expire" the withdrawal & liquidation, respectively.

Signed-off-by: Nick Pai <npai.nyc@gmail.com>

### CODE REVIEW

File: core/scripts/local/AdvanceEMP.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    for (const address of contractAddresses) {
+      // The governor is always registered as a contract, but it isn't an ExpiringMultiParty.
+      if (address !== Governor.address) {
+        emp = await ExpiringMultiParty.at(address);
+
+        // Advance time
+        let leapForward = argv.time ? argv.time : 3601;
+        let currentTime = await emp.getCurrentTime();
+        let newTime = toBN(currentTime.toString()).add(toBN(leapForward.toString()));
+        await emp.setCurrentTime(newTime.toString());
```
Comment (chrismaree) [inline core/scripts/local/AdvanceEMP.js]: here you are only advancing time within the emp. this _should_ be fine but if in testing a vote needs to happen on price (dispute, settlement, emergencyshutdown) then the DVM time will be disjoint from the emp time. Quick, dirty fix is to set the DVM time here as well. Longer term solution is outlined in #1047

File: core/scripts/local/AdvanceEMP.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/**
+ * This script advances time in the EMP forward by some specified amount of seconds (or one hour and one second by default).
+ *
+ * `truffle exec ./scripts/local/AdvanceEMPTime --time 10000`
```
Comment (ptare) [inline core/scripts/local/AdvanceEMP.js]: I think you're taking in time as seconds, so this example should be `10`.

File: core/scripts/local/AdvanceEMP.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const Registry = artifacts.require("Registry");
+const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
+
+// Contracts we need to interact with.
+let registry;
+let emp;
+
+/** ***************************************************
+ * Main Script
+ /*****************************************************/
```
Comment (ptare) [inline core/scripts/local/AdvanceEMP.js]: Nothing to change here, but personally I find these comments to be clutter. I know we use them in a lot places so totally fine to leave it in.

File: core/scripts/local/AdvanceEMP.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        let currentTime = await emp.getCurrentTime();
+        let newTime = toBN(currentTime.toString()).add(toBN(leapForward.toString()));
+        await emp.setCurrentTime(newTime.toString());
+        currentTime = await emp.getCurrentTime();
+        let currentTimeReadable = new Date(Number(currentTime.toString()) * 1000);
+        console.log(`Set time to ${currentTimeReadable} for the EMP @ ${emp.address}`);
+      }
+    }
+  } catch (err) {
+    console.error(err);
```
Comment (ptare) [inline core/scripts/local/AdvanceEMP.js]: Use callback(err) to terminate the script with an error. We're not perfect about this in other scripts, which we should eventually clean up.

File: core/scripts/local/AdvanceEMP.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    // Query all registered EMP's.
+    for (const address of contractAddresses) {
+      // The governor is always registered as a contract, but it isn't an ExpiringMultiParty.
+      if (address !== Governor.address) {
+        emp = await ExpiringMultiParty.at(address);
+
+        // Advance time
+        let leapForward = argv.time ? argv.time : 3601;
+        let currentTime = await emp.getCurrentTime();
+        let newTime = toBN(currentTime.toString()).add(toBN(leapForward.toString()));
```
Comment (ptare) [inline core/scripts/local/AdvanceEMP.js]: I think all of these variables besides `currentTime` can all be const.

File: core/scripts/local/AdvanceEMP.js (hunk 1 above)
Comment (nicholaspai) [reply core/scripts/local/AdvanceEMP.js]: I've been using the mock oracle, but I suppose I can grab the relevant oracle and set its time forward too.

File: core/scripts/local/AdvanceEMP.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+const argv = require("minimist")(process.argv.slice(), { string: ["time"] });
+
+// Deployed contract ABI's and addresses we need to fetch.
+const Governor = artifacts.require("Governor");
+const Registry = artifacts.require("Registry");
+const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
+
+// Contracts we need to interact with.
+let registry;
+let emp;
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply core/scripts/local/AdvanceEMP.js]: yep

File: core/scripts/local/AdvanceEMP.js (hunk 2 above)
Comment (nicholaspai) [reply core/scripts/local/AdvanceEMP.js]: ah yes

File: core/scripts/local/AdvanceEMP.js (hunk 3 above)
Comment (nicholaspai) [reply core/scripts/local/AdvanceEMP.js]: yeah actually I agree. I like the top line comment that explains the purpose of the script but I don't need to annotate the main script which obviously is the one and only function

File: core/scripts/local/AdvanceEMP.js (hunk 4 above)
Comment (nicholaspai) [reply core/scripts/local/AdvanceEMP.js]: would this still print out the err?

Comment (nicholaspai) [reply core/scripts/local/AdvanceEMP.js]: it does, done.

File: core/scripts/local/AdvanceEMP.js (hunk 5 above)
Comment (nicholaspai) [reply core/scripts/local/AdvanceEMP.js]: yep youre right

File: core/scripts/local/AdvanceEMP.js (hunk 1 above)
Comment (chrismaree) [reply core/scripts/local/AdvanceEMP.js]: cool. it should just advance all time consistently I think

**Pull request merged**